### PR TITLE
Split contract creation & funding

### PR DESCRIPTION
**Pull request by harshjv**

### CODE REVIEW

File: src/worker/errorHandler.js [hunk 1]
```diff
@@ -10,17 +10,15 @@ const getResponseBody = e => _.get(e, 'response.data') || _.get(e, 'response.bod
...
-
-  if (err.code === 'ECONNREFUSED') {
-    job.schedule('in ' + config.worker.httpConnectionErrorDelay)
-    return job.save()
-  }
-
   if (
     err.name === 'RescheduleError' ||
-    err.name === 'PossibleTimelockError'
+    ( // do not retry PossibleTimelockError in production
```
Comment (monokh) [inline src/worker/errorHandler.js:15]: I think we should expand the check to any jobs that involve transactions, it's hard to predict where issues will be and I can't see why we need to retry them. Possibly something sometimes fails and it needs manual action but that's much better than retries happening in error.

Comment (monokh) [reply src/worker/errorHandler.js:15]: Have i got it wrong? See a lot of has changed,  this means we only retry on specific errors yes?

Comment (harshjv) [reply src/worker/errorHandler.js:15]: We only retry in `production` if `TxNotFoundError`, `PendingTxError`, `BlockNotFoundError`, which takes place before we send a transaction. (guard conditions)

File: src/worker/jobs/fund-swap.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    await order.log('FUND_SWAP', null, {
+      fromBlock: fromCurrentBlockNumber
+    })
+
+    return agenda.now('find-refund-tx', { orderId: order.orderId, fromLastScannedBlock: fromCurrentBlockNumber })
+  }
+
+  const toSecondaryFundTx = await order.fundSwapTransaction()
+  if (toSecondaryFundTx) {
```
Comment (monokh) [inline src/worker/jobs/fund-swap.js:46]: I hope it's not possible that `fundSwapTransaction` will return null for ERC20

Comment (monokh) [reply src/worker/jobs/fund-swap.js:46]: also I think the name should be `fundSwap`?

**Pull request merged**